### ISSUE TITLE
Upgrading listen currently forces a Ruby upgrade on its users

### ISSUE THREAD

**Issue by Confusion** [guard/listen#409]
Since version 3.1.2, your `listen.rb` includes
````
require 'ruby_dep/warning'                                                      
RubyDep::Warning.new.show_warnings  
````
This suggests it just causes warnings, but in fact it causes a `bundle update` to fail with
````
Gem::InstallError: ruby_dep requires Ruby version >= 2.2.5, ~> 2.2.
````
if you are e.g. on Ruby 2.2.4. We are fully aware of the problems with Ruby 2.2.4, but at the moment we are stuck with it. We would nevertheless like to be able to upgrade our gem dependencies and the 'show_warnings' shown above prevents that. 

Would it be possible to move this code to a test/development/qa part of your gem? Or depend on a different version of ruby_dep with more lenient dependencies? Do you really only support 2.2.5+?

**Comment by e2** [comment]
You can downgrade Listen to 3.0.x or even to the 2.x branch - which doesn't require RubyDep.

A later version of Bundler (prerelease?) could maybe do this automatically (downgrade to earlier Listen).

If you have any issues with the 3.0.x branch, let me know.

**Comment by Confusion** [comment]
I solved it by downgrading to 3.1.1 exactly, but it took quite a while to figure out that that would solve the problem and we could improve that. But since I now see you are also the author of ruby_dep, let's take the discussion over there.
